pkg/apis/numaflow/v1alpha1/source.go

```diff
@@ -36,8 +36,6 @@ type Source struct {
 	// +optional
 	Nats *NatsSource `json:"nats,omitempty" protobuf:"bytes,4,opt,name=nats"`
 	// +optional
-	RedisStreams *RedisStreamsSource `json:"redisStreams,omitempty" protobuf:"bytes,5,opt,name=redisStreams"`
-	// +optional
 	UDTransformer *UDTransformer `json:"transformer,omitempty" protobuf:"bytes,6,opt,name=transformer"`
```

Comment:
Update the index?

Reply:
Good catch. But just thinking - For Protobuf, in general, you're not supposed to modify existing indices for backwards compatibility purposes. Does that no longer apply for the Protobuf we generate here? (I'm not really sure exactly how Protobuf is being used in this K8S context - but I suppose as long as the "recipient" is also using this same version then we're fine.)

Reply:
I don't think any of our users are using protobuf encoding to communicate to k8s, it should be safe to adjust the index.

Reply:
The answer to `how Protobuf is being used in this K8S context` - 

https://kubernetes.io/docs/reference/using-api/api-concepts/#alternate-representations-of-resources

Reply:
Thanks for sending! In that case, it seems like if something were to use Protobof it would be something which has a running client that can interpret it, right? I wonder if we would want to have backward compatibility for them, or if they should basically always just adapt to the latest version of the CRD? (I suppose it's a general question for whether to allow changing numbers going forward)

Reply:
No one always keeps backward compatibility, projects like k8s have a deprecation process (e.g. mark the fields as deprecated, and remove them in N releases - the numbers are also updated).

Reply:
Did you want to follow that process at some point in the future? I know we haven't even released v1.0 yet, so as far as I understand we're not worrying about backward compatibility much right now. I am okay to update the numbers if you prefer.

Reply:
We should have similar process later on. Thanks!